mantidproject/mantid
# Context Menu Change Axis' Scale to Axis Aware on Workbench

~~Merge after #26655~~

**Description of work.**
when changing the scale on a figure with multiple plots, right-clicking and changing the axis now only changes the axis you clicked on rather than all of them.

**To test:**
- Run the below script to produce a figure containing 4 axes (data to load attached).
- Then right-click on any of the sets of axes and select Axes → Lin x/Log y.
- Only the axes clicked on should have their y-axis changed to a log scale.
[GEM38370_Focussed.zip](https://github.com/mantidproject/mantid/files/3495979/GEM38370_Focussed.zip)
```
# The following line helps with future compatibility with Python 3
# print must now be used as a function, e.g print('Hello','World')
from __future__ import (absolute_import, division, print_function, unicode_literals)

# import mantid algorithms, numpy and matplotlib
from mantid.simpleapi import *
from mantid.api import AnalysisDataService as ADS

import matplotlib.pyplot as plt

import numpy as np

Load(Filename=r'GEM38370_Focussed.nxs', OutputWorkspace='GEM38370_Focussed')
workspace = ADS.retrieve('GEM38370_Focussed')

fig = plt.figure(figsize=(11, 8))
for i in range(1, 5):
    ax = fig.add_subplot(2, 2, i, projection='mantid')
    ax.plot(workspace, specNum=i+1)
plt.show()
```

Fixes #26629.

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
